src/language-yaml/print/mapping-item.js

```diff
@@ -149,6 +153,29 @@ function printMappingItem(path, options, print) {
   ]);
 }
 
+function isMappingKeyBreak(key, options) {
+  if (options.proseWrap === "never") {
+    return false;
+  }
+
+  if (!key || !key.content) {
+    return false;
+  }
+
+  const { content } = key;
+
+  if (isNode(content, ["flowMapping", "flowSequence"])) {
+    return false;
+  }
+
+  if (options.proseWrap === "preserve") {
+    return content.position.start.line !== content.position.end.line;
+  }
+
+  const keyLength = content.position.end.offset - content.position.start.offset;
+  return keyLength > options.printWidth;
```

Comment:
This is incorrect, since it can be a child, which means it will be less room than `printWidth`.

Also, this case absolutely should not in "// force explicit Key" branch.

Reply:
Maybe exclude from "// force singleline" branch.